gluten-core/src/main/scala/org/apache/spark/sql/execution/ColumnarShuffleExchangeExec.scala

```diff
@@ -129,7 +129,7 @@ case class ColumnarShuffleExchangeExec(
   }
 
   override def getShuffleRDD(partitionSpecs: Array[ShufflePartitionSpec]): RDD[ColumnarBatch] = {
-    cachedShuffleRDD
+    new ShuffledColumnarBatchRDD(columnarShuffleDependency, readMetrics, partitionSpecs)
```

Comment:
@ulysses-you Why need this change?

Reply:
It's the key change to use AQEShuffleRead directly. AQEShuffleRead will call getShuffleRDD to get the custom shuffle RDD. See:
https://github.com/apache/spark/blob/9cbc2d138a1de2d3ff399d224d817554ba0b9e18/sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEShuffleReadExec.scala#L250C1-L265